perf-tool/src/agentOptions.cpp

```diff
@@ -0,0 +1,281 @@
+
+#include <jvmti.h>
+#include <ibmjvmti.h>
+#include "infra.hpp"
+#include <string>
+#include <cstring>
+#include <jvmti.h>
+#include <stdio.h>
+#include <iostream>
+#include <string>
+#include <unistd.h>
+
+#include "agentOptions.hpp"
+#include "infra.hpp"
+#include "monitor.hpp"
+#include "objectalloc.hpp"
+#include "exception.hpp"
+#include "methodEntry.hpp"
+
+#include "json.hpp"
+
+using json = nlohmann::json;
+
+void invalidCommand(std::string function, std::string command)
+{
+    printf("Invalid command with parameters: {functionality: %s, command: %s}\n", function.c_str(), command.c_str());
+}
+
+void invalidFunction(std::string function, std::string command)
+{
+    printf("Invalid function with parameters: {functionality: %s, command: %s}\n", function.c_str(), command.c_str());
+}
+
+void invalidRate(std::string function, std::string command, int sampleRate)
+{
+    printf("Invalid rate with parameters: {functionality: %s, command: %s, sampleRate: %i}\n", function.c_str(), command.c_str(), sampleRate);
+}
+
+void modifyMonitorEvents(std::string function, std::string command, int rate)
+{
+    jvmtiCapabilities capa;
+    jvmtiError error;
+    setMonitorSampleRate(rate);
+    (void)memset(&capa, 0, sizeof(jvmtiCapabilities));
+    error = jvmti->GetCapabilities(&capa);
+    check_jvmti_error(jvmti, error, "Unable to get current capabilties");
+    if (capa.can_generate_monitor_events)
+    {
+        if (!command.compare("stop"))
+        {
+            (void)memset(&capa, 0, sizeof(jvmtiCapabilities));
+            capa.can_generate_monitor_events = 1;
+
+            error = jvmti->RelinquishCapabilities(&capa);
+            check_jvmti_error(jvmti, error, "Unable to relinquish \n");
+            error = jvmti->SetEventNotificationMode(JVMTI_DISABLE, JVMTI_EVENT_MONITOR_CONTENDED_ENTERED, (jthread)NULL);
+            check_jvmti_error(jvmti, error, "Unable to disable MonitorContendedEntered event.");
+        }
+        else
+        { // c == start
```

Comment:
It's better to validate that you indeed have received a start command.Similarly, in other parts of the code.